accessibility-checker-engine/src/v2/aria/ARIAMapper.ts

```diff
@@ -653,7 +675,15 @@ export class ARIAMapper extends CommonMapper {
             "article": true,
             "aside": true,
             "nav": true,
-            "section": true
+            "section": true,
+            "main": true
```

Comment:
This is an interesting point. @tombrunet we might need to chat about this or have another location for this. The reason we did this was to cover this case: ([[link](https://www.w3.org/TR/html-aria/)](url))
![image](https://user-images.githubusercontent.com/4451128/145459090-f03070b0-f8dc-4493-9b02-ad04b5a4e1af.png)

Reply:
@carmacleod Why is 'main' included in this list for ARIA in HTML? It feels weird to have sectioning content + main. The object and variables are correct for the context it's being used. Maybe the variable name is wrong. I'm not sure what a better name might be though. `sectioningContentPlus`?

Reply:
@tombrunet @aliunwala
I will try to find out the rationale behind header/footer needing to be top-level landmarks (i.e. in the context of `body`).
I suspect it will have to do with the fact that there _should_ only be one of each, and people expect them to be outside of the main content. I also suspect, looking at this further, that `form` and `search` may be missing from the "If not a descendant of ..." list for header/footer in HTML-ARIA. Heh, maybe even header & footer themselves. :) I will ask Scott.

In the meantime, maybe just add a comment to the definition of sectioningContent saying that for the purpose of ensuring that header and footer are top-level landmarks, main is included in this list.

Reply:
@tombrunet @aliunwala
For completeness on the `form` and `search` investigations:
- **form**: apparently a `form` can sometimes wrap the whole page, and you'd want to allow having banner/contentinfo landmarks inside the form (although it would be weird to make the wrapping form a landmark by adding a label).
- **search**: if/when we have an [HTML `<search>` element](https://github.com/whatwg/html/issues/5811), then we can consider adding search element to the header/footer implicit landmark scoping rules, but there are currently no restrictions on banner/contentinfo landmarks within elements with role=search.

Note that I think I managed to come up with a closer-to-reality name for `sectioningContent` and `sectioningRoles`. How about `scopingContent` and `scopingRoles`? Is that more understandable? Feel free to add a comment that these lists are basically for scoping implicit landmark role assignment for HTML header & footer elements.